packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -96,6 +96,7 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 	border-radius: 2px;
 	background: ${ COLORS.theme.background };
 	border: 1px solid ${ COLORS.theme.foreground };
+	z-index: 9999; // Ensure the popover is on top
```

Comment:
| Before | After |
|---------|---------|
|  <img width="185" alt="Screenshot 2024-06-07 at 12 07 05 p m" src="https://github.com/WordPress/gutenberg/assets/81248/6d09b986-d4f2-40fc-a911-c02dca5afa0a"> |  <img width="199" alt="Screenshot 2024-06-07 at 12 04 22 p m" src="https://github.com/WordPress/gutenberg/assets/81248/c1764cc6-56d9-4e38-b5df-1d6c0ae68f40"> |

Reply:
Since the original `V1` implementation used the pre-determinded `z-index` for popovers, we should at least use the same `z-index` value — here's what we did in `DropdownMenuV2`:

https://github.com/WordPress/gutenberg/blob/c94f863b5df5034524e86490427864779b12754e/packages/components/src/dropdown-menu-v2/styles.ts#L76-L78

Reply:
Opened https://github.com/WordPress/gutenberg/pull/62821 to work on the select popover fixes separately